### DISCUSSION TITLE
Reading device wifi signal strength

### DISCUSSION THREAD

**Discussion by andnic** [jasonacox/tinytuya#384] [Q&A]
I've just gotten a script working to log temperature data from several thermostats which is working very smoothly.

I'd like to also log the wifi signal strength just for monitoring the health of my network, has anyone been able to read this? It's available under "Device Information" in the app, but I haven't been able to divine how to retrieve it with tinytuya.

Thanks!

**Comment by uzlonewolf** [comment]
I'm not aware of any way to get this data locally via the LAN.  Presumably there is a Cloud API that can pull it, but I'd need to do some digging to try and find it.

**Comment by ramzcode** [comment]
Any update on the idea ?

**Reply by jasonacox** [reply]
I have not seen any way to do this locally. It does seem that it is being measured and sent to the cloud. You can connect to the Tuya cloud to pull this. https://developer.tuya.com/en/docs/cloud/2287954993?id=Kdqf1hiwhdc7f
